test/modules/storage-aws-s3/snapshot_create_test.go

```diff
@@ -26,41 +29,64 @@ import (
 )
 
 func Test_S3Storage_StoreSnapshot(t *testing.T) {
+	testCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
 	testdataMainDir := "./testData"
 	testDir := makeTestDir(t, testdataMainDir)
```

Comment:
I think that we can also refactor `makeTestDir` and make use of `t.TempDir()` feature